examples/python/self-reflection-optimization-patterns.py

```
@@ -0,0 +1,556 @@
+"""
+Self-Reflection Optimization Patterns Example
+
+This example demonstrates advanced self-reflection capabilities including adaptive strategies,
+quality-based termination, multi-agent reflection coordination, and performance optimization.
+"""
+
+import json
```

Ruff (0.12.2)

`json` imported but unused

Remove unused import: `json`

(F401)